pkg/client/client.go

```diff
@@ -48,15 +81,21 @@ func (c *ContainerGroupsClientWrapper) CreateCG(ctx context.Context, resourceGro
 		return err
 	}
 
-	result, err := c.CGClient.CreateOrUpdateSender(addReq)
+	err = retry.OnError(retry.DefaultBackoff,
+		func(err error) bool {
+			return ctx.Err() == nil
+		}, func() error {
+			result, err := c.CGClient.CreateOrUpdateSender(addReq)
+			if err != nil {
+				err = autorest.NewErrorWithError(err, "containerinstance.ContainerGroupsClient", "CreateOrUpdate", result.Response(), "Failure sending request trying again")
+				return err
+			}
+			return err
+		})
 	if err != nil {
-		err = autorest.NewErrorWithError(err, "containerinstance.ContainerGroupsClient", "CreateOrUpdate", result.Response(), "Failure sending request")
-		return err
-	}
-	if result.Response().StatusCode != http.StatusOK {
```

Comment:
The API is not consistent with the returned status core, it has multi values (200, 201, 0 or nil)